### DISCUSSION TITLE
usb://0 Cannot connect to crazyflie 2 using USB

### DISCUSSION THREAD

**Discussion by reginaldStjohn** [bitcraze/discussions#939] [General]
We are trying to change our CrazyFlie address using the instructions to connect with the usb cable. We get the usb://0 showing up in the dropdown box and our virtual machine shows the crazyflie connected to the USB. However, when we hit connect we get the following error:

INFO:cflib.drivers.cfusb:Looking for devices....
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.
INFO:cflib.crazyflie:Callback->Connection initialized[usb://0]
INFO:cflib.drivers.cfusb:Looking for devices....
ERROR:cflib.crazyflie:Couldn't load link driver: The device has no langid (permission issue, no string descriptors supported or device error)

**Comment by krichardsson** [comment]
Which VM are you using? 
We released a new version a couple of days ago (2023.10), I just tested and this works.

**Reply by reginaldStjohn** [reply]
When I say "we", I mean one of our team. We have 4 sets of crazyflies and radios.  On one computer we had an image from earlier this year. Several others have downloaded the newest (2023.10) image so we all have slightly different versions.  

We had one crazyflie of which we did not know the address so we could not connect to it with the radio.
We eventually just used one of the "working", new images to connect to it via USB so we could see what the address was set to.

NOTE: The newest version of the VM does not seem to work in our host of choice, VMWare.  It never can get network access so we cannot access anything outside the VM.  If we run the image in VBox then we get network.

**Reply by knmcguire** [reply]
The VM image is built for virtual box and we don't test it with the release on VMware... so this might need some additional changes to the configuration. Is there a reason why you'd like rather to work with VMware instead of virtualbox?

This error that you posted, was that an error in vmware or in virtualbox? Could you check the permissions of usb devices, maybe it's missing the USB permissions for the crazyflie:

https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/installation/usb_permissions/

**Reply by reginaldStjohn** [reply]
The error we posted was in VMWare.  We use VMWare at work because our IT department has licenses for it to create virtual machines and access to the paid "Workstation" version that has some options the player does not have.

We had used it with some of the previous versions of the Bitcrazy VM image without problems so we just tried the new image and had the network difficulties.

We would have to investigate the USB problem some more.  It may be related to the older VM image. At this point in time we have a work around so it it not urgent for us.

Thank you.

**Reply by knmcguire** [reply]
ah oke good to know. We don't have a license to VMware so we won't be able to help out there... but at least it is not blocking you anymore.
